File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        int x = 0, y = 1, profit = 0;

        if (prices.length <= 1) return 0;

        // x = 1, y = 2
        while (x < prices.length-1 && y < prices.length) {
            if (prices[x] >= prices[y]) {
                x = y;
            } else {
                profit = Math.max(profit, prices[y] - prices[x]);
            }
            y++;
        }
        return profit;
    }
}

File: Data Structures & Algorithms/maximum-product-subarray/submission-0.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    maxProduct(nums: number[]): number {
        let maxProd = -Infinity
        for (let i = 0 ; i < nums.length ; i++) {
            let currProd = nums[i]
            maxProd = Math.max(maxProd, currProd)
            for (let j = i+1 ; j < nums.length ; j++) {
                currProd *= nums[j]
                maxProd = Math.max(currProd, maxProd)
            }
        }

        return maxProd
    }
}
